### ISSUE TITLE
[Bug]: String issues with railtype GRF cause crash

### ISSUE THREAD

**Issue by 2TallTyler** [OpenTTD/OpenTTD#13147]
### Version of OpenTTD

Nightly, 8d394c6

### Expected result

NewGRF track sets cannot cause crashes, even if they make mistakes with string handling.

### Actual result

[JP+ Tracks](https://github.com/OpenTTD-JPplus/JPplusTracks) v0.3.0, in some configurations, shows nonsense in the rail toolbar title and can even cause a crash.

![Screenshot 2024-12-03 083423](https://github.com/user-attachments/assets/5a457ac1-66bd-4b93-a691-c89e15c5d565)

(To be clear, I'm not accusing the JP+ team of doing anything wrong, for all I know this could be an OpenTTD bug!)

### Steps to reproduce

To reproduce the string issue, set the GRF parameters to this (there are more if you scroll down, but I didn't change those). Then start a game and open the default railtype, `Standard Gauge Track (Low-Grade)`.

![parameters](https://github.com/user-attachments/assets/bc9f7bc8-0261-441e-b837-2cca37af071b)

Or just load this savegame: [ReproductionSave.zip](https://github.com/user-attachments/files/17993706/ReproductionSave.zip)

I have also experienced a crash when clicking on the main toolbar to open the rail construction menu. It is inconsistent, but here's the crash files: [crash20241203133224.zip](https://github.com/user-attachments/files/17993698/crash20241203133224.zip)

**Comment by 2TallTyler** [comment]
@PeterN, should this have been closed by #13153?

**Comment by PeterN** [comment]
No, It fixes the crash but not the string issues that lead to the crash.

**Comment by PeterN** [comment]
Resolved by various things leading up to #13285
